detekt-rules-style/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/BracesOnIfStatements.kt

```diff
@@ -0,0 +1,136 @@
+package io.gitlab.arturbosch.detekt.rules.style
+
+import io.gitlab.arturbosch.detekt.api.CodeSmell
+import io.gitlab.arturbosch.detekt.api.Config
+import io.gitlab.arturbosch.detekt.api.Debt
+import io.gitlab.arturbosch.detekt.api.Entity
+import io.gitlab.arturbosch.detekt.api.Issue
+import io.gitlab.arturbosch.detekt.api.Rule
+import io.gitlab.arturbosch.detekt.api.Severity
+import io.gitlab.arturbosch.detekt.api.config
+import io.gitlab.arturbosch.detekt.api.internal.Configuration
+import org.jetbrains.kotlin.com.intellij.psi.PsiElement
+import org.jetbrains.kotlin.psi.KtBlockExpression
+import org.jetbrains.kotlin.psi.KtExpression
+import org.jetbrains.kotlin.psi.KtIfExpression
+import org.jetbrains.kotlin.psi.KtWhenExpression
+import org.jetbrains.kotlin.psi.psiUtil.siblings
+
+/**
+ * This rule detects `if` statements which do not comply with the specified rules.
+ * Keeping braces consistent would improve readability and avoid possible errors.
+ *
+ * <noncompliant>
+ * val i = 1
+ * if (i > 0)
+ *    println(i)
+ * else {
+ *    println(-i)
+ * }
+ * </noncompliant>
+ *
+ * <compliant>
+ * || singleLine = 'never' multiLine = 'always'
+ * val x = if (condition) 5 else 4
+ * val x = if (condition) {
+ *     5
+ * } else {
+ *     4
+ * }
+ * </compliant>
+ */
+class BracesOnIfStatements(config: Config = Config.empty) : Rule(config) {
+
+    override val issue = Issue(
+        javaClass.simpleName,
+        Severity.Style,
+        "Braces do not comply with the specified policy",
+        Debt.FIVE_MINS
+    )
+
+    @Configuration("single-line braces policy")
+    private val singleLine: String by config(BRACES_NEVER)
+
+    @Configuration("multi-line braces policy")
+    private val multiLine: String by config(BRACES_ALWAYS)
```

Comment:
```suggestion
    private val multiLine: BracePolicy by config<String>(BRACES_ALWAYS) {
        when(it) {
            "never" -> BracePolicy.Never
            "consistent" -> BracePolicy.Consistent
            "always" -> BracePolicy.Always
            else -> error("Unknown value $it, the allowed values are: ...")
        }
    }
```

(I'm writing this from memory so maybe this is not the correct syntax but I know that something like this could be done